UVtools.Core/Operations/OperationRepairLayers.cs

```diff
@@ -403,6 +406,64 @@ void initImage()
                 });
             }
 
+
+            if (_repairSuctionCups)
+            {
+                (bool canDrill, Point location) GetDrillLocation(LayerIssue issue, int diameter)
+                {
+                    var centroid = EmguCV.EmguContour.GetCentroid(new VectorOfPoint(issue.Contours[0]));
```

Comment:
centroid is not safe... If a suction is inside a ring like the bearing model, the centeroi  will be the center of the bearing, but suction is traped on the outter ring. 
Eg:
![UVtools_2021-09-17_01-44-26](https://user-images.githubusercontent.com/113281/133706238-10e3f9e7-fd04-455b-99f2-18fc22d5b163.png)

Centroid of hallow part is on center of the model, which will make a not valid position.
Maybe here we can use Distance transformation

Reply:
This is accounted for in the code by later ensuring that the centroid is actually in the contour area, this is achieved immediately below this where we draw the contour, and the planned drilling circle and ensure overlap. If there is no overlap we do not repair it.

Reply:
Using distance transformation and find the most itensity pixel will always ensure a center inside the issue, you can relate to centroid too. But can be a improvement for latter

![UVtools_2021-09-17_01-56-31](https://user-images.githubusercontent.com/113281/133706992-0ffc954a-bbc0-457d-a565-8b0497d725a1.png)